src/tableView.js

```diff
@@ -364,6 +383,49 @@ class TableView {
     return Math.min(this.instance.columnIndexMapper.getRenderableIndexesLength(), this.settings.maxCols);
   }
 
+  /**
+   * Returns the number of renderable rows.
+   *
+   * @returns {number}
+   */
+  countRenderableRows() {
+    return Math.min(this.instance.rowIndexMapper.getRenderableIndexesLength(), this.settings.maxRows);
+  }
+
+  /**
+   * Returns number of not hidden records counting from the passed starting index.
+   * The counting direction can be controlled by `incrementBy` argument.
+   *
+   * @param {number} visualIndex The visual index from which the counting begins.
+   * @param {string} axix The axis as 'row' or 'column'.
+   * @param {number} incrementBy If `-1` then counting is backwards or forward when `1`.
+   * @returns {number}
+   */
+  countNotHiddenRecords(visualIndex, axix, incrementBy) {
+    if (isNaN(visualIndex)) {
+      return 0;
+    }
+
+    const indexMapper = this.instance[`${axix}IndexMapper`];
+    const firstVisibleIndex = indexMapper.getFirstNotHiddenIndex(visualIndex, incrementBy);
+    const renderableIndex = indexMapper.getRenderableFromVisualIndex(firstVisibleIndex);
+
+    if (!Number.isInteger(renderableIndex)) {
+      return 0;
+    }
+
+    let notHiddenRows = 0;
+
+    if (incrementBy < 0) {
+      // Zero-based numbering acts here as a count of not hidden rows.
```

Comment:
I would change that to: "Zero-based numbering **for renderable indexes** ~~acts here as a count~~ corresponds to a number of not hidden ~~rows~~ indexes."